roseus/roseus.cpp

```diff
@@ -1089,7 +1089,7 @@ pointer ROSEUS_WAIT_FOR_SERVICE(register context *ctx,int n,pointer *argv)
 
   int32_t timeout = -1;
 
-  if( n > 1 )
+  if( n > 1 && argv[1] != NIL && argv[1] > 0)
```

Comment:
1. `argv[1]` is of pointer type, so `argv[1]>0` will always be true except for the nullptr.
```
(gdb) p makeint(-1)
$1 = (struct cell *) 0xfffffffffffffffe
(gdb) p (makeint(-1)>0)
$2 = 1
```
2. A negative timeout value can often mean "wait indefinitely", so I am not sure if ignoring it is correct.
3. On a separate note, `WAIT_FOR_SERVICE` forces an integer timeout, but there is no reason to do that, since `ros::Duration` accepts float values as well. ref: https://github.com/Affonso-Gui/jsk_roseus/commit/2f220a4051d8429e62db33cecb32345cb3aabe88

Reply:
Borderline related to https://github.com/jsk-ros-pkg/jsk_roseus/pull/623

Reply:
what i want to this in PR is checking nil.
checking negative is just an optional thing.
if you think we dont need it, i can remove it.

Reply:
For checking `nil`, I think this PR correctly solve the issue...

current master
```
2.irteusgl$ (ros::wait-for-service "/hoge" nil)
Call Stack (max depth: 20):
  0: at (ros::wait-for-service "/hoge" nil)
  1: at #<compiled-code #X56455ffad280>
/home/knorth55/install/jskeus/eus/Linux64/bin/irteusgl 0 error: integer expected in (ros::wait-for-service "/hoge" nil)
```

this PR.

```
2.irteusgl$ (ros::wait-for-service "/hoge" nil)
[ INFO] [1669892226.285970439]: waitForService: Service [/hoge] has not been advertised, waiting...

```